urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/wikipedia/Main.java

```diff
@@ -0,0 +1,221 @@
+import wikipedia.utils.URLUtils;
+
+public class Main
+{
+	final static PrintStream ps = System.out;
+
+	final static String wikipedia_url = "http://en.wikipedia.org/w/index.php";
+	final static String wikipedia_args = "?action=render&printable=yes&title=";
+
+	final static String article_file_path = "/wiki/";
+	final static int articles_max = 5000;
+	static String articles_counter_file = "/WEB-INF/properties/articles_counter.properties";
+	static Properties articles_prop;
+
+	/**
+	 * Get the article as XML from Wikipedia
+	 * 
+	 * @param article_title
+	 * @return
+	 * @throws IOException 
+	 */
+	private static String get_xmlFromWikipedia(String article_title) throws IOException
+	{
+		String encodedArticle = null;
+		
+		try
+		{
+			encodedArticle = URLEncoder.encode(article_title, "UTF-8");
+		} catch (UnsupportedEncodingException e)
+		{
+			e.printStackTrace();
+		}
+		
+		URLConnection conn = URLUtils.getURLConnection(wikipedia_url + wikipedia_args + encodedArticle);
+		
+
+		return URLUtils.getSource(conn);
+	}
+
+	/**
+	 * Save the XML file
+	 * 
+	 * @param file_name
+	 * @param content
+	 */
+	private static void save_xmlFile(String file_name, String content)
+	{
+		BufferedWriter bw = null;
+		try
+		{
+
+			bw = new BufferedWriter(new FileWriter(file_name));
+			bw.write(content);
+
+		} catch (IOException exc)
+		{
+			// Logger.getLogger (Main.class.getName ()).log (Level.SEVERE, null,
+			// exc);
+			exc.printStackTrace();
+		} finally
+		{
+
+			try
+			{
+				if (bw != null)
+					bw.close();
+			} catch (IOException e)
+			{
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+	}
+
+	/**
+	 * Get total amount of articles on own server
+	 * 
+	 * @return
+	 */
+	private static int get_totalArticles()
+	{
+		try
+		{
+			FileInputStream propInFile = new FileInputStream(WebSocketServer.path
+					+ articles_counter_file);
+			articles_prop = new Properties();
+
+			articles_prop.load(propInFile);
+			String articles = articles_prop.getProperty("articles", "0");
+
+			return Integer.valueOf(articles).intValue();
+		} catch (IOException e)
+		{
+			e.printStackTrace();
+			ps.println("I/O failed.");
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Delete one random XML
+	 */
+	private static void delete_oneArticle()
+	{
+		File folder = new File(WebSocketServer.path + article_file_path);
+		File[] listOfFiles = folder.listFiles();
+
+		for (int i = 0; i < listOfFiles.length; i++)
+		{
+			if (listOfFiles[i].isFile() && listOfFiles[i].canWrite())
+			{
+				if (listOfFiles[i].getName().contains(".xml"))
+				{
+					if (listOfFiles[i].delete())
+						break;
+					else
+						throw new IllegalArgumentException("Delete failed");
+				}
+			}
+		}
+	}
+
+	/**
+	 * Update the property file with the articles amount
+	 * 
+	 * @param articles_total
+	 */
+	private static void update_totalArticles(int articles_total)
+	{
+		try
+		{
+			FileOutputStream propOutFile = new FileOutputStream(WebSocketServer.path
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *propOutFile*. Execution paths that do not contain closure statements were detected. Ensure *propOutFile* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).